src/components/files/createDefaultFolders.ts

```diff
@@ -0,0 +1,57 @@
+// Creates (and populates if appropriate) the folders that are not saved as part
+// of git repo. We run this for safety when starting the server, or before
+// saving or loading a snapshot.
+
+import {
+  PREFERENCES_FOLDER,
+  PREFERENCES_FILE_NAME,
+  LOCALISATION_FOLDER,
+  FILES_FOLDER,
+  SNAPSHOT_FOLDER,
+  BASE_SNAPSHOT_FOLDER,
+  GENERIC_THUMBNAILS_FOLDER,
+  GENERIC_THUMBNAILS_SOURCE_FOLDER,
+  DATABASE_FOLDER,
+  BASE_SNAPSHOT_NAME,
+} from '../../constants'
+import fs from 'fs'
+import path from 'path'
+import { execSync } from 'child_process'
+import { makeFolder } from '../utilityFunctions'
+
+export function createDefaultDataFolders() {
+  try {
+    makeFolder(SNAPSHOT_FOLDER, 'Creating SNAPSHOTS folder')
+    // Copy core_templates to snapshots folder
+    execSync(`cp -r '${DATABASE_FOLDER}/${BASE_SNAPSHOT_NAME}' '${SNAPSHOT_FOLDER}'`)
+  } catch {
+    console.log('\nProblem creating SNAPSHOTS folder\n')
+  }
+
+  try {
+    if (!fs.existsSync(FILES_FOLDER)) {
+      makeFolder(FILES_FOLDER, 'Creating FILES folder')
+      execSync(`cp -r '${BASE_SNAPSHOT_FOLDER}/files/.' '${FILES_FOLDER}'`)
+      // Restore generic thumbnails, they get wiped out during snapshot loading
+      makeFolder(path.join(GENERIC_THUMBNAILS_FOLDER))
+      execSync(`cp -r '${GENERIC_THUMBNAILS_SOURCE_FOLDER}/.' '${GENERIC_THUMBNAILS_FOLDER}'`)
+    }
+  } catch {
+    console.log('\nProblem creating FILES folder\n')
+  }
+
+  // If localisation folder is missing, we assume it's a fresh install and
+  // fetch them (and preferences) from core templates
+  try {
+    if (
+      !fs.existsSync(LOCALISATION_FOLDER) ||
+      !fs.existsSync(path.join(LOCALISATION_FOLDER, 'languages.json'))
+    ) {
+      makeFolder(LOCALISATION_FOLDER, 'Restoring LOCALISATIONS and PREFERENCES')
+      execSync(`cp -r '${BASE_SNAPSHOT_FOLDER}/localisation/.' '${LOCALISATION_FOLDER}'`)
```

Comment:
I can't actually remove "preferences" from the repo root like I did with "localisations" because nothing will run without a prefs file. However, the preferences will get replaced with the "core_templates" one whenever "localisations" is restored -- that way we can be sure that whatever is in "core_templates"  is the "official" source of truth and we can change the main one around without worrying about changing defaults.